stories/modules-tagmanager-settings.stories.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Tag Manager Module Settings Stories.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { storiesOf } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME as CORE_SITE, AMP_MODE_PRIMARY, AMP_MODE_SECONDARY } from '../assets/js/googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../assets/js/googlesitekit/datastore/user';
+import { SettingsMain as TagManagerSettings } from '../assets/js/modules/tagmanager/components/settings';
+import * as fixtures from '../assets/js/modules/tagmanager/datastore/__fixtures__';
+import { STORE_NAME, ACCOUNT_CREATE } from '../assets/js/modules/tagmanager/datastore/constants';
+import { createTestRegistry, freezeFetch } from '../tests/js/utils';
+import createLegacySettingsWrapper from './utils/create-legacy-settings-wrapper';
+
+const Settings = createLegacySettingsWrapper( 'tagmanager', TagManagerSettings );
+
+const defaultSettings = {
+	accountID: '',
+	containerID: '',
+	internalContainerID: '',
+	ampContainerID: '',
+	internalAMPContainerID: '',
+	useSnippet: true,
+};
+
+storiesOf( 'Tag Manager Module/Settings', module )
+	.addDecorator( ( storyFn ) => {
```

Comment:
Nice! So this function runs initially and then calls the actual story function? May be worth refactoring other stories too, although fairly low priority.

Reply:
Yeah, decorators are intended for "wrapping" your stories. They can be defined at various levels, but this will run for each story in this `storiesOf` call. I'm not sure if it's intended to be used in this way to pass a value to the story function but it works great 😄 .